### PR TITLE
feat(server): support sort by metadata.creationTimestamp for listing

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

As suggested in the title

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cd1989** [comment]
Such implementation makes simple things completed, and prone to errors. I'd prefer:

```
type ReposSortByNameAsc []*HarborRepo

func (r ReposSortByNameAsc) Len() int {
	return len(r)
}

func (r ReposSortByNameAsc) Swap(i, j int) {
	r[i], r[j] = r[j], r[i]
}

func (r ReposSortByNameAsc) Less(i, j int) bool {
	return r[i].Name < r[j].Name
}
```

And use it as:

```
sort.Sort(ReposSortByNameAsc(repos))
```

**Comment by zhujian7** [comment]
@supereagle @cd1989 PTAL

File: pkg/server/handler/v1alpha1/integration.go [hunk 1]
```diff
@@ -38,7 +40,7 @@ func ListIntegrations(ctx context.Context, tenant string, query *types.QueryPara
 	}
 
 	items := secrets.Items
-	integrations := []api.Integration{}
+	integrations := make([]api.Integration, 0)
```
Comment (cd1989) [inline pkg/server/handler/v1alpha1/integration.go]: `var integrations []api.Integration`

File: pkg/server/types/common.go [hunk 2]
```diff
@@ -7,10 +7,10 @@ type QueryParams struct {
 	Start  int64  `source:"query,start,default=0"`
 	Limit  int64  `source:"query,limit,default=99999"`
 	Filter string `source:"query,filter"`
-	// SortBy represents the order of results. For example:
-	// creationTime: desc order by creationTime
-	// -creationTime: asc order by creationTime
-	SortBy string `source:"query,sortBy"`
+	// Sort will sorts the resoults by metadata.creationTimestamp
```
Comment (cd1989) [inline pkg/server/types/common.go]: `resoults ` --> `results`

**Comment by zhujian7** [comment]
@cd1989 PTAL

**Pull request merged**